client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -246,6 +250,47 @@ public String getDeviceId() {
         }
     }
 
+    /**
+     * Checks whether the command is targeted at a gateway.
+     * <p>
+     * This is the case when the commands got redirected and hence
+     * the device id (ie. the gateway id in that case) is different
+     * from the original device id.
+     *
+     * @return {@code true} if the device id is a gateway id.
+     * @throws IllegalStateException if this command is invalid.
+     */
+    public boolean isTargetedAtGateway() {
+        if (isValid()) {
+            final String originalDeviceId = getOriginalDeviceId();
+            return originalDeviceId != null && !originalDeviceId.equals(getDeviceId());
+        } else {
+            throw new IllegalStateException("command is invalid");
+        }
+    }
+
+    /**
+     * Gets the device identifier used in the original command.
```

Comment:
IMHO the comment should explicitly explain that the value returned is extracted from the AMQP message's *to* property.